cibyl/sources/elasticsearch/api.py

```diff
@@ -185,17 +186,29 @@ def get_deployment(self, **kwargs):
             query=query_body
         )
 
+        IP_REGEX = re.compile("ipv(.)")
         job_objects = {}
         for hit in hits:
             job_name = hit['_source']['jobName']
             url = hit['_source']['envVars']['JOB_URL']
             # If the key exists assign the value otherwise assign unknown
             topology = hit.get('_source', {}).get('envVars', {}).get(
                 "JP_IRVIRSH_TOPOLOGY_NODES", "unknown")
-            ip_version = hit.get('_source', {}).get('envVars', {}).get(
-                "JP_OSPD_NETWORK_PROTOCOL", "unknown")
             release = hit.get('_source', {}).get('envVars', {}).get(
                 "JP_OSPD_PRODUCT_VERSION", "unknown")
+            ip_version = hit.get('_source', {}).get('envVars', {}).get(
+                "JP_OSPD_NETWORK_PROTOCOL", "unknown")
+
+            if ip_version != 'unknown':
+                matches = IP_REGEX.search(ip_version)
+                ip_version = matches.group(1)
+
+            # Check if necessary filter by IP version:
+            if 'ip_version' in kwargs:
+                ip_version_argument = kwargs.get('ip_version').value
+                if ip_version_argument and \
+                        ip_version_argument[0] != ip_version:
```

Comment:
I would change the second check to `ip_version not in ip_version_argument`.

Reply:
Could be a good logic, but in that case it wouldn't filter for all the cases, it will work just to v4 and v6 independently.

Reply:
You mean in the case the user specifies `--ip-version 4 6`? In that case it would find both kinds of jobs and add them

Reply:
Yes, in that case it works for both. Do we need to show also `unknown` ip address when we pass `--ip-address` argument? Or just the jobs that have one of both versions setted?

Reply:
I've changed the condition.